exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java

```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
   @Override
   public Object defaultValue() { return defaultValue; }
 
+  @Override
+  public void setDefaultFromString(String value) {
```

Comment:
This is such a handy operation that you might want to split it into two:

Object valueFromString(String value)

And your current function, which would call the above function internally.